### PR TITLE
TRON-76: Change signTransaction return type

### PR DESCRIPTION
**Pull request by argjv**
[TRON-76](https://bitgoinc.atlassian.net/browse/TRON-76) Fix pending approvals and align with the return types of other coins

### CODE REVIEW

File: modules/core/src/v2/baseCoin.ts [hunk 1]
```diff
@@ -160,13 +160,15 @@ export interface AddressCoinSpecific {
 
 export interface FullySignedTransaction {
   txHex: string;
+  transaction?: any; // full transaction in JSON format
```
Comment (tylerlevine) [inline modules/core/src/v2/baseCoin.ts]: Please use txHex property as we discussed.

File: modules/core/src/v2/baseCoin.ts [hunk 2]
```diff
@@ -160,13 +160,15 @@ export interface AddressCoinSpecific {
...
 }
 
 export interface HalfSignedTransaction {
   halfSigned?: {
     txHex?: string;
     payload?: string;
     txBase64?: string;
+    transaction?: any; // full transaction in JSON format
```
Comment (tylerlevine) [inline modules/core/src/v2/baseCoin.ts]: Please use txHex property as we discussed.

File: modules/core/src/v2/coins/trx.ts [hunk 3]
```diff
@@ -16,6 +17,9 @@ import {
   SignTransactionOptions,
   VerifyAddressOptions,
   VerifyTransactionOptions,
+  TransactionFee,
+  TransactionRecipient as recipient,
```
Comment (tylerlevine) [inline modules/core/src/v2/coins/trx.ts]: Types should begin with a capital letter.

File: modules/core/src/v2/coins/trx.ts [hunk 4]
```diff
@@ -26,17 +30,17 @@ export interface TronSignTransactionOptions extends SignTransactionOptions {
   prv: string;
 }
 
-export interface TransactionPrebuild {
-  txHex: string;
-  txid: string;
+export interface txInfo {
```
Comment (tylerlevine) [inline modules/core/src/v2/coins/trx.ts]: Types should begin with a capital letter

File: modules/core/src/v2/coins/trx.ts [hunk 5]
```diff
@@ -137,13 +141,25 @@ export class Trx extends BaseCoin {
   }
 
   signTransaction(params: TronSignTransactionOptions): SignedTransaction {
-    const tx = params.txPrebuild.transaction;
+    const tx = params.txPrebuild.txInfo.transaction;
     // this prv should be hex-encoded
-    const halfSigned = tronWeb.utils.crypto.signTransaction(params.prv, tx);
-    if (_.isEmpty(halfSigned.signature)) {
+    const transaction = tronWeb.utils.crypto.signTransaction(params.prv, tx);
+    if (_.isEmpty(transaction.signature)) {
```
Comment (twtaylorbitgo) [inline modules/core/src/v2/coins/trx.ts]: it's possible this has a signature on it and is invalid. (so a fully signed tx coming through this flow would have 1 sig). this depends on previous state. we should check if it came in unsigned, does it have 1 sig? and then if it was half, does it have 2?

Comment (argjv) [reply modules/core/src/v2/coins/trx.ts]: I'm not sure how useful this is going to be since I would expect the builder to check for that, but I'll add it.

**Pull request merged**